### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by shihabrafio**
In this PR, included
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+
```
Comment (Olamarx) [inline classroom.rb]: - [ ] The students should be readable by adding attr_reader :students to it.

File: student.rb [hunk 2]
```diff
@@ -1,9 +1,12 @@
 require_relative 'Person'
 
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(age, classroom, name = 'Unknown', parent_permission: true)
     super(name, parent_permission, age)
     @classroom = classroom
+    classroom.students << self
```
Comment (Olamarx) [inline student.rb]: - [ ] There should be a method that serves as the 'belongs-to' side (a student belongs to a classroom), it should be responsible for adding the student if it is not there before. More like this:

```
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :author
```
Comment (Olamarx) [inline book.rb]: - [ ] rentals should also have the getter and setter accessibility. You can just include it in the attr_accessor.

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (Olamarx) [inline rental.rb]: - [ ] The book and person should not be read only, it should be a setter too so I will recommend adding it to the attr_accessor.

File: person.rb [hunk 5]
```diff
@@ -11,6 +11,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
     super()
   end
```
Comment (Olamarx) [inline person.rb:17]: - [ ] I will encourage you to add rentals for the purpose of setter and setter by using the attr_accessor

**Review by Olamarx — APPROVED** [review]
Hi @shihabrafio,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Commit messages meaningful ✔️
- Readme file professional ✔️
- Readable codebase ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Olamarx** [comment]
@shihabrafio approval was a mistake, I wanted to request changes, I would do that on dashboard. Apologies.

**Review by Davidosky007 — APPROVED** [review]
Hi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Well written PR 
No linter errors 
Good looking readme 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**